docs/extending-entry-points.rst

```diff
@@ -0,0 +1,224 @@
+Extending papermill through entry points
+========================================
+
+What are entry points?
+----------------------
+
+The python packaging documentation describes `entry points`_ as:
+
+    Entry points are a mechanism for an installed distribution to advertise
+    components it provides to be discovered and used by other code. For example:
+
+    Distributions can specify console_scripts entry points, each referring to a
+    function. When pip (or another console_scripts aware installer) installs the
+    distribution, it will create a command-line wrapper for each entry point.
+
+    Applications can use entry points to load plugins; e.g. Pygments (a syntax
+    highlighting tool) can use additional lexers and styles from separately
+    installed packages. For more about this, see Creating and discovering
+    plugins.
+
+When running, papermill looks for `entry points`_ that implement input / output
+(I/O) handlers, and execution handlers.
+
+Developing new I/O handlers
+---------------------------
+
+Virtually the first thing that happens when papermill is used is that the input
+notebook is read in. This is managed by I/O handlers, which allow papermill to
+access not just the local filesystem, but also remote services such as Amazon
+S3. The same goes for writing the executed notebook to a file system: I/O
+handlers allow papermill to write files to S3 or otherwise.
+
+Creating a new handler
+~~~~~~~~~~~~~~~~~~~~~~
+
+Writing your own I/O handler simply requires writing a class that has four
```

Comment:
```suggestion
Writing your own I/O handler requires writing a class that has four
```
I'd avoid words like "simply" or "just do X" on the one hand it makes it sounds like it should be simple (and if you don't find it simple you are not smart enough) and on the other hand it annoys the anti-[weasel word](https://en.wikipedia.org/wiki/Weasel_word) people.

Reply:
Agreed! I've updated this here and elsewhere.